### DISCUSSION TITLE
How to exit VR mode?

### DISCUSSION THREAD

**Discussion by paulmelis** [NVlabs/instant-ngp#1215] [Q&A]
I haven't found a way to exit VR mode, once you're in it, other than restarting instant-ngp. Is there a better way?

**Comment by Tom94** [comment]
Hi, when you take off your headset, instant-ngp should automatically return to single-view rendering (at which point you can re-enter VR by putting on the headset again).

Does this not happen for you? If so, which headset do you have?

**Reply by paulmelis** [reply]
It's a Varjo XR-3. Just tried it, taking it off doesn't switch back to normal mode. Might be a nice idea to have a controller button exit to non-VR mode.

**Reply by Tom94** [reply]
Thanks for the quick reply! I unfortunately don't have a Varjo at hand for testing, so can't troubleshoot why it doesn't automatically detect when being removed from your head.

I added a button to disconnect from the headset in any case (updated binaries should be ready in about 1h) and I also added a dropdown to change the environment blending mode for good measure (I can't test it with my available headsets, but presumably the Varjo can benefit from compositng the environment in).

**Reply by paulmelis** [reply]
The disconnect button works fine, thanks! The dropdown for the blend mode is a bit weird. It contains (with the XR-3 here) a `BlendOpaque` entry and a `H?` entry. The first one works as expected, but I don't know where the second one's name comes from. It also doesn't change the blending as far as I can tell (can still see the checkered sky dome).

Edit: and I don't have a dev environment set up here to set `m_print_environment_blend_modes ` to `true` to see what gets reported ;-)

Edit 2: Ah, [OpenXR Explorer](https://github.com/maluoi/openxr-explorer) does list the supported blend modes on the XR-3: 

![image](https://user-images.githubusercontent.com/1408287/216549481-51940e27-9e0f-4df4-b2d0-01f035a07570.png)

Edit 3 (sorry for the mess): the 2nd entry's name is not consistent over different runs (`H?`, `?`, `??`), so could be an invalid memory access of some sort. I *do* see a change in skydome rendering, btw, but only after switching away from blendopaque and back to it again. The skydome then turns black.

**Reply by Tom94** [reply]
Whoops -- I'll fix the dropdown, my bad. In the meantime: does the black sky dome turn transparent if you disable DLSS (after switching back and forth in the dropdown)?

**Reply by paulmelis** [reply]
No, DLSS yes/no makes no difference in BlendOpaque mode

**Reply by Tom94** [reply]
Thanks for checking. Unfortunately, I don't have any bright ideas about what else could be broken. I'll get back to mixed reality once I have a local test setup. (The dropdown is fixed on latest `master` in any case.)

**Reply by paulmelis** [reply]
Okay, with your latest updates the Blend mode now works!
